### DISCUSSION TITLE
GitHub auth

### DISCUSSION THREAD

**Discussion by jbcaillau** [control-toolbox/control-toolbox.github.io#31] [General]
@PierreMartinon on HTTPS / SSH, check this (both should work provided the services are up - I can’t imagine a server without HTTP but I had one without SSH for some time [^1]):

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github#authenticating-with-the-command-line

https://cli.github.com/manual/gh_auth_login

https://cli.github.com/manual/gh_auth_login

[^1]: currently using HTTPS with works fine with `gh`

**Comment by PierreMartinon** [comment]
Yup, I solved it by declaring the registry with a ssh url instead of a https one.
The problem is a recent one: last year (?) github stopped authorizing push commands when using basic https (username/password) authentification (you can still clone and pull, but not push).

Earlier I also had the complication of using two distinct github accounts, which requires two keys (you cannot share a key between accounts) and some configuration, but works too.

The surprising part is that the register() command worked for Olivier even though the registry was added with a http url. Maybe this has something to do with some global authentification preferences on his machine, that I didn't have with the multiple account settings. Who knows :D

I never tried the token thingie but it could be a solution too.

Note that the register() issue is a bit more tricky than standard git login since the push is called automatically, so you have less control over the options. However the LocalRegistry doc actually explains how to handle ssh auth and/or private repos, so I guess it would be enough to mention the ssh case and link to the doc.

**Comment by ocots** [comment]
I have changed our doc on the [wiki](https://github.com/control-toolbox/control-toolbox.github.io/wiki/Dev-setup):

```bash
pkg> registry add git@github.com:control-toolbox/ct-registry.git
     Cloning registry from "git@github.com:control-toolbox/ct-registry.git"
       Added registry `ct-registry` to `~/.julia/registries/ct-registry`
```

**Reply by jbcaillau** [reply]
I have re-included the `HTTP` command line on the [wiki](https://github.com/control-toolbox/control-toolbox.github.io/wiki/Dev-setup): both work.
